### PR TITLE
Fusion: migrate colorbleed fork code

### PR DESCRIPTION
**Pull request by jakubjezek001**
## Brief description
Integrating Colorbleed's work and adding `Set Framerange` to menu items.

## Description
Original PR description here https://github.com/BigRoy/OpenPype/pull/2
- added `Set Framerange` to menu items to set framerange to comp from asset document data.

### CODE REVIEW

**Comment by mkolar** [comment]
Task linked: [OP-2667 Fusion: set framerange from menu button](https://app.clickup.com/t/6658547/OP-2667)

**Comment by BigRoy** [comment]
Thanks for setting this up. This is coming from my messy `colorbleed` branch where I continuously pull in development streams and make live for testing in our production. I'm a bit wary about seeing all those commits in here but having the "merged output" at the end only contain the changes for Fusion. Because that would mean if I'd pull this it'd overwrite my local commits without telling me there's a merge conflict (since you had already 'resolved' that merge).

Any `git` master here to know how to squash this and avoid that issue?

File: openpype/hosts/fusion/deploy/MenuScripts/install_pyside2.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# This is just a quick hack for users running Py3 locally but having no
+# Qt library installed
+import os
+
+try:
+    print("Qt library found, nothing to do.")
+    from Qt import QtWidgets
```
Comment (iLLiCiTiT) [inline openpype/hosts/fusion/deploy/MenuScripts/install_pyside2.py:7]: I would do the print after import.
```suggestion
    from Qt import QtWidgets
    print("Qt library found, nothing to do.")
```

File: openpype/hosts/fusion/hooks/pre_fusion_setup.py [hunk 2]
```diff
@@ -14,101 +15,36 @@ class FusionPrelaunch(PreLaunchHook):
...
+        prefs = os.path.join(HOST_DIR, "deploy", "fusion_shared.prefs")
+        self.log.info(f"Setting {pref_var}: {prefs}")
+        self.launch_context.env[pref_var] = prefs
 
-                # copy file or directory from our folders to fusion's folder
-                if not os.path.isdir(src):
-                    shutil.copy2(src, dst)
-                else:
-                    shutil.copytree(src, dst)
+        __import__("avalon.fusion")
```
Comment (iLLiCiTiT) [inline openpype/hosts/fusion/hooks/pre_fusion_setup.py:49]: Module `avalon.fusion` does not exists anymore and it does not make sense to import it in prelaunch hook. Import of `pyblish` also does not make sense here.

**Comment by iLLiCiTiT** [comment]
> This currently DOES contain all the other changes. We should extract only the Fusion changes from our repo.
Best might be to "cherry pick" only those commits related to the fusion code (with a bit of luck I have those commits completely separate).

I should have read this first. If you can cherry pick only commits that are fusion related then that would be best. But since it seems that it is combined I would suggest copy paste changes you want to merge and commit just that. You can commit changes in "reasonable chunks" so the commits make sense that's up to you.

From my experience it is easier to copy and redo commits then cherry pick. Spend less time and nerves.

**Comment by jakubjezek001** [comment]
> > I should have read this first. If you can cherry pick only commits that are fusion related then that would be best. But since it seems that it is combined I would suggest copy paste changes you want to merge and commit just that. You can commit changes in "reasonable chunks" so the commits make sense that's up to you.
> 
> Yes, that sounds about right. We could just take whatever is in the "Fusion" folder and other possible 'fusion related changes' and copy those changes over onto `develop` I think. With a PR set up with just that commit it'll be much easier to go over the changelog.

I could do it myself but only that would be assigning ownership to myself which would not be fair.

**Comment by BigRoy** [comment]
Thanks @jakubjezek001 - closing this in favor of #2769. Let me know if I forgot to include something important from your commits.

**Comment by jakubjezek001** [comment]
> Thanks @jakubjezek001 - closing this in favor of #2769. Let me know if I forgot to include something important from your commits.

Awesome, thanks @BigRoy - I was looking into #2769 and it seems to be having all of my changes regarding setFramerange.

**Pull request closed without merging**